extensions/rich_text_components/Image/directives/oppia-noninteractive-image.component.ts

```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the Image rich-text component.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ImageDimensions, ImagePreloaderService } from 'pages/exploration-player-page/services/image-preloader.service';
+import { AssetsBackendApiService } from 'services/assets-backend-api.service';
+import { ContextService } from 'services/context.service';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { AppConstants } from 'app.constants';
+import { downgradeComponent } from '@angular/upgrade/static';
+@Component({
+  selector: 'oppia-noninteractive-image',
+  templateUrl: './image.component.html',
+  styleUrls: []
+})
+export class NoninteractiveImage implements OnInit {
+  @Input() filepathWithValue: string;
+  @Input() altWithValue: string = '';
+  @Input() captionWithValue: string = '';
+  filepath: string;
+  imageUrl: string = '';
+  imageAltText: string = '';
+  imageCaption: string = '';
+  loadingIndicatorUrl;
+  isLoadingIndicatorShown: boolean = false;
+  isTryAgainShown: boolean = false;
+  dimensions: ImageDimensions;
+  imageContainerStyle: {height: string, width: string};
+  loadingIndicatorStyle: {height: string, width: string};
```

Comment:
```suggestion
  imageContainerStyle: { height: string, width: string };
  loadingIndicatorStyle: { height: string, width: string };
```
Maybe can be separated into its own type?

Reply:
Done!